gobblin-modules/gobblin-kafka-09/src/main/java/org/apache/gobblin/kafka/client/Kafka09ConsumerClient.java

```diff
@@ -162,6 +170,35 @@ public KafkaConsumerRecord apply(ConsumerRecord<K, V> input) {
     });
   }
 
+  @Override
+  public Map<String, Metric> metrics() {
+    Map<MetricName, KafkaMetric> kafkaMetrics = (Map<MetricName, KafkaMetric>) this.consumer.metrics();
+    Map<String, Metric> codaHaleMetricMap = new HashMap<>();
+
+    kafkaMetrics
+        .forEach((key, value) -> codaHaleMetricMap.put(canonicalMerticName(value), kafkaToCodaHaleMetric(value)));
+    return codaHaleMetricMap;
+  }
+
+  /**
+   * Convert a {@link KafkaMetric} instance to a {@link Metric}.
+   * @param kafkaMetric
+   * @return
+   */
+  private Metric kafkaToCodaHaleMetric(final KafkaMetric kafkaMetric) {
```

Comment:
Isn't this likely to be re-used in other implementation under the same package? I am suggesting to make it package private.

Reply:
Good point. Unfortunately, the KafkaMetric class is specific to a given Kafka version, making code duplication necessary. So it seems that each consumer client version will have its own implementation of the method.